### PR TITLE
SAA-1469 3rd party dependency updates and small code changes to help resolve security warnings. Also resolves some warnings and a failing test.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Note going forwards if you want to check/analyse for updates locally you will need have a local environment variable set.

See the DPS gradle docs [here](https://github.com/ministryofjustice/dps-gradle-spring-boot/blob/main/release-notes/5.x.md) on point 5.11 for more details.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -136,6 +136,7 @@ workflows:
           slack_channel: << pipeline.parameters.alerts-slack-channel >>
           context:
             - hmpps-common-vars
+          cache_key: "v3_0"
```
Comment (matt-crowson-moj) [inline .circleci/config.yml:139]: This is needed as part of the update, refer to gradle boot plugin for details from 5.11 on-wards.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/ActivityScheduleIntegrationTest.kt [hunk 2]
```diff
@@ -474,7 +474,7 @@ class ActivityScheduleIntegrationTest : IntegrationTestBase() {
       .expectBody(typeReference<LinkedHashMap<String, Any>>())
       .returnResult().responseBody!!
 
-    assertThat(response["content"] as List<ActivityCandidate>).hasSize(5)
+    assertThat((response["content"] as List<*>).asListOfType<ActivityCandidate>()).hasSize(5)
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/ActivityScheduleIntegrationTest.kt:477]: Resolves warning.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/WaitingListApplicationIntegrationTest.kt [hunk 3]
```diff
@@ -102,7 +103,7 @@ class WaitingListApplicationIntegrationTest : IntegrationTestBase() {
     results["empty"] isEqualTo false
     results["totalElements"] isEqualTo 5
 
-    val content = results["content"] as List<LinkedHashMap<String, Any>>
+    val content = (results["content"] as List<*>).asListOfType<LinkedHashMap<String, Any>>()
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/WaitingListApplicationIntegrationTest.kt:106]: Resolves warning.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/common/CollectionExt.kt [hunk 4]
```diff
@@ -6,3 +6,10 @@ fun <T, R> Collection<T>.ifNotEmpty(block: (Collection<T>) -> R): R? =
 fun <T> Collection<T>.containsAny(otherCollection: Collection<T>): Boolean {
   return this.intersect(otherCollection.toSet()).isNotEmpty()
 }
+
+/**
+ * Conversion is not guaranteed, only use when you know (or at least expect) all types to be the same type.
+ */
+inline fun <reified T> List<*>.asListOfType(): List<T> =
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/common/CollectionExt.kt:13]: Put casting in one place to resolve warnings.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityServiceTest.kt [hunk 5]
```diff
@@ -875,8 +875,9 @@ class ActivityServiceTest {
 
   @Test
   fun `updateActivity - update end date`() {
-    val updateActivityRequest: ActivityUpdateRequest = mapper.read("activity/activity-update-request-4.json")
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityServiceTest.kt:878]: This file was using a fixed date (now in the past) ergo it started to fail without any code changes.  Had to fix to get changes in.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/config/JwtConfig.kt [hunk 6]
```diff
@@ -10,7 +10,7 @@ import java.security.interfaces.RSAPublicKey
 
 @Configuration
 class JwtConfig {
-  private val keyPair: KeyPair
+  private lateinit var keyPair: KeyPair
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/config/JwtConfig.kt:13]: Resolves a warning.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/common/CollectionExt.kt [hunk 7]
```diff
@@ -6,3 +6,9 @@ fun <T, R> Collection<T>.ifNotEmpty(block: (Collection<T>) -> R): R? =
 fun <T> Collection<T>.containsAny(otherCollection: Collection<T>): Boolean {
   return this.intersect(otherCollection.toSet()).isNotEmpty()
 }
+
+/**
+ * Conversion is not guaranteed, only use when you know (or at least expect) all types to be the same type.
+ */
+@Suppress("UNCHECKED_CAST")
+inline fun <reified T> List<*>.asListOfType() = this as List<T>
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/common/CollectionExt.kt:14]: Put suppressed casting in one place to resolve warnings.

File: .circleci/config.yml (hunk 1 above)
Comment (timharrison-moj) [reply .circleci/config.yml:139]: Do we all need to generate our own API keys and set this in our local env vars as NVD_API_KEY?

Comment (timharrison-moj) [reply .circleci/config.yml:139]: How does circle-ci provide this additional env var?

Comment (matt-crowson-moj) [reply .circleci/config.yml:139]: No mention of it in the release notes of the dps gradle boot plugin, just that we need the cache_key adding.

The notes themselves don't go into any more details.

**Pull request merged**